src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -292,60 +293,25 @@ public void HandleWrapperException(Exception exception)
 
         public Stream TryGetStreamInjector(Stream stream, Encoding encoding, string contentType, string requestPath)
         {
-            if (stream == null)
-            {
-                return null;
-            }
-
-            if (encoding == null)
-            {
-                return null;
-            }
-
-            if (contentType == null)
-            {
-                return null;
-            }
-
-            if (requestPath == null)
-            {
-                return null;
-            }
-
-            try
-            {
-                var transaction = _transactionService.GetCurrentInternalTransaction();
-                if (transaction == null)
-                {
-                    return null;
-                }
-
-                var shouldInject = _browserMonitoringPrereqChecker.ShouldAutomaticallyInject(transaction, requestPath, contentType);
-                if (!shouldInject)
-                {
-                    return null;
-                }
+            var script = TryGetRUMScriptInternal(contentType, requestPath);
+            return script == null ? null : new BrowserMonitoringStreamInjector(() => script, stream, encoding);
+        }
 
-                // Once the transaction name is used for RUM it must be frozen
-                transaction.CandidateTransactionName.Freeze(TransactionNameFreezeReason.AutoBrowserScriptInjection);
-                var script = _browserMonitoringScriptMaker.GetScript(transaction, null);
-                if (script == null)
-                {
-                    return null;
-                }
+        public async Task TryInjectBrowserScriptAsync(string contentType, string requestPath, byte[] buffer, Stream baseStream, ITransaction transaction)
```

Comment:
Should baseStream be checked for null? Or is the null possibility just an asp.net thing?

Reply:
I honestly doubt `null` was even a possibility in asp.net. Looks to me like some unnecessary defensive coding.